components/sinks/cu_zenoh_ros_sink/src/lib.rs

```diff
@@ -0,0 +1,190 @@
+mod attachment;
+mod error;
+mod keyexpr;
+mod liveliness;
+mod node;
+mod topic;
+
+use attachment::encode_attachment;
+use cdr::{CdrBe, Infinite};
+use cu29::clock::RobotClock;
+use cu29::prelude::*;
+use cu_ros_payloads::RosMsgAdapter;
+use error::cu_error_map;
+use liveliness::{node_liveliness, publisher_liveliness};
+use node::Node;
+use topic::Topic;
+use zenoh::bytes::Encoding;
+use zenoh::Config;
+
+use std::marker::PhantomData;
+
+// Only one node per session
+const NODE_ID: u32 = 0;
+
+// Only one publisher per session
+const PUBLISHER_ID: u32 = NODE_ID + 1;
+
+/// This is a sink task that sends ROS-compatible messages to a Zenoh topic.
+/// P is the payload type of the messages, which must be convertible to ROS format.
+/// Hence the payload type must implement the `RosMsgAdapter` trait.
+pub struct ZenohRosSink<P>
+where
+    P: CuMsgPayload + RosMsgAdapter,
+{
+    _marker: PhantomData<P>,
+    config: ZenohRosConfig,
+    ctx: Option<ZenohRosContext>,
+}
+
+pub struct ZenohRosConfig {
+    session: zenoh::Config,
+    domain_id: u32,
+    namespace: String,
+    node: String,
+    topic: String,
+}
+
+pub struct ZenohRosContext {
+    session: zenoh::Session,
+    publisher: zenoh::pubsub::Publisher<'static>,
+    // Token have to be kept alive (undeclared on drop)
+    #[allow(dead_code)]
+    node_token: zenoh::liveliness::LivelinessToken,
+    #[allow(dead_code)]
+    publisher_token: zenoh::liveliness::LivelinessToken,
+    sequence_number: u64,
+}
+
+impl<P> Freezable for ZenohRosSink<P> where P: CuMsgPayload + RosMsgAdapter {}
+
+impl<'cl, P> CuSinkTask<'cl> for ZenohRosSink<P>
+where
+    P: CuMsgPayload + RosMsgAdapter + 'cl + 'static,
```

Comment:
while correct rust, this 'static feels out of place. Conceptually those types only need to live for the duration of one CopperList ie. one execution cycle.